spec/index.bs

```diff
@@ -251,6 +251,155 @@ dictionary GPUObjectDescriptorBase {
 </script>
 
 
+Programming Model {#programming-model}
+======================================
+
+## Timelines ## {#programming-model-timelines}
+
+*This section is non-normative.*
+
+A computer system with a user agent at the front and GPU at the back
+has components working on different timelines in parallel:
+
+: <dfn dfn>Content timeline</dfn>
+:: Associated with the execution of the Web script.
+    It includes calling all methods described by this specification.
+
+: <dfn dfn>Device timeline</dfn>
+:: Associated with the GPU device operations
+    that are issued by the user agent. It includes creation of GPU resources
+    and state objects, which are typically synchronous operations from the point
+    of view of the user agent part that controls the GPU,
+    but it can live in a separate OS process.
+
+: <dfn dfn>Queue timeline</dfn>
+:: Associated with the execution of operations
+    on the compute units of the GPU. It includes actual rendering draw
+    and compute jobs that run on the GPU.
+
+In this specification, asynchronous operations are used when the result value
+depends on work that happens on any timeline other than the [=Content timeline=].
+They are represented by callbacks and promises in JavaScript.
+
+<div class="example">
+{{GPUComputePassEncoder/dispatch(x, y, z)|GPUComputePassEncoder.dispatch()}}:
+
+  1. User encodes {{GPUComputePassEncoder}}, which happens on the [=Content timeline=].
```

Comment:
"Encoding" a `GPUComputePassEncoder`, I'd rather say "Encode a dispatch command with the ...". Also the actual encoding happens at the earliest on the device timeline no? I think an example with just `GPUQueue.submit` would work, where the call itself is on the application timeline, and the rest stays the same as in your example.